### PR TITLE
optimize the contract status judgment logic

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

File: libprecompiled/ContractStatusPrecompiled.cpp [hunk 1]
```diff
@@ -169,29 +170,31 @@ void ContractStatusPrecompiled::freeze(
...
-        }
-        else if (ContractStatus::Nonexistent == status)
-        {
-            result = CODE_TABLE_AND_ADDRESS_NOT_EXIST;
-        }
-        else
-        {
-            result = updateFrozenStatus(context, tableName, STATUS_TRUE, origin);
-        }
+        result = CODE_TABLE_AND_ADDRESS_NOT_EXIST;
```
Comment (bxq2011hust) [inline libprecompiled/ContractStatusPrecompiled.cpp]: don't use other precompiled's error code

File: libprecompiled/ContractStatusPrecompiled.cpp [hunk 2]
```diff
@@ -96,34 +96,35 @@ bool ContractStatusPrecompiled::checkPermission(
...
-            status = ContractStatus::Available;
-        }
+        return ContractStatus::AddressNonExistent;
     }
-    else
-    {  // valid address but nonexistent
-        status = ContractStatus::Nonexistent;
+
+    auto codeHashEntries = table->select(storagestate::ACCOUNT_CODE_HASH, table->newCondition());
+    if (codeHashEntries->size() > 0 &&
```
Comment (bxq2011hust) [inline libprecompiled/ContractStatusPrecompiled.cpp]:
```suggestion
    if (
```

File: libprecompiled/ContractStatusPrecompiled.cpp [hunk 3]
```diff
@@ -96,34 +96,35 @@ bool ContractStatusPrecompiled::checkPermission(
...
+        toHex(EmptySHA3) == codeHashEntries->get(0)->getField(storagestate::STORAGE_VALUE))
+    {
+        return ContractStatus::NotContractAddress;
     }
 
-    if (ContractStatus::Invalid == status)
+    auto frozenEntries = table->select(storagestate::ACCOUNT_FROZEN, table->newCondition());
+    if (frozenEntries->size() > 0 &&
```
Comment (bxq2011hust) [inline libprecompiled/ContractStatusPrecompiled.cpp]:
```suggestion
    if (
```

File: libprecompiled/ContractStatusPrecompiled.cpp (hunk 2 above)
Comment (chaychen2005) [reply libprecompiled/ContractStatusPrecompiled.cpp]: removed

**Pull request merged**